### PR TITLE
Add taxation guide

### PR DESCRIPTION
**Pull request by mamhoff**
This adds a guide for pricing and taxation, two related and difficult to grasp areas. It should be merged after https://github.com/solidusio/solidus/pull/1598.

### CODE REVIEW

File: guides/pricing_and_taxation.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+Taxation and prices are two intertwined topics in Solidus, mainly because a large number of jurisdictions in the world use VAT-style taxation, which influences the price at which an item is sold.
+
+## Taxation
+
+Solidus represents taxes for an order's items by using `Spree::TaxCategory` and `Spree::TaxRate`.
+
+Tax rates are associated with a tax category and a zone. If an order's `tax_address` (usually the shipping address) falls within a zone with tax rates, those rates will apply to all line items and shipments with respectively matching tax categories.
+
+When a tax rate gets applied to a line item, two things happen: A tax adjustment is created on the line item, and the line item's `additional_tax_total` and `included_tax_total` are changed to reflect the newly created adjustments. The amounts are then added to the order's `*_tax_total` fields.
```
Comment (bbuchalter) [inline guides/pricing_and_taxation.md]: What's the difference between `additional_tax_total` and `included_tax_total`? Perhaps link down to the lower section? Or does the more conceptual, introductory section "Sales tax vs. Value-Added Tax" belong higher up in this document?

File: guides/pricing_and_taxation.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+### Tax Categories
+
+Products within Solidus can be linked to Tax Categories, which are then used find the taxation rate for the products when they are purchased. One Tax Category can be set to being the default for the entire system, which means that if a product doesn't have a tax category set, then the default tax category would be used.
+
+The Solidus default is to treat everything as exempt from tax. Only if a product has a tax category set, or there is a global default tax category, will products be taxed. The tax category is specific to taxation, and is normally never seen by the user so you could call it something generic like "Taxable Goods." If you wish to tax certain products at different rates, however, then you will want to choose something more descriptive (ex. "Clothing.").
+
+### Tax Rates
+
+A tax rate is essentially a percentage amount charged based on the net price. Tax rates also contain other important information.
```
Comment (bbuchalter) [inline guides/pricing_and_taxation.md]: I think the phrase "net price" merits further explanation. Does that include item-level adjustments? Order-level adjustments?

File: guides/pricing_and_taxation.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+### Sales tax vs. Value-Added Tax
+
+In e-Commerce, the type of tax we care most about is consumption tax, of which two main models are found: Sales Tax and Value Added Tax (VAT).
+
+The Sales tax policies found in the USA are *additional* taxes. They do not change the amount of a line item, but are added just before calculating the order total. For Sales tax, the prices for items do not change when adding them to an order.
+
+With Value-Added Tax (VAT), as found in Europe, but also in many other parts of the world, the prices of products and line items need to be shown *including* the consumption tax. When a product is exported, the price of that product has to change: It will be lower, because there is no included tax anymore.
+
+***
+Most taxes can be considered one of these two types. For instance, in Australia customers pay a Goods and Services Tax (GST). This is basically equivalent to VAT in Europe.
```
Comment (bbuchalter) [inline guides/pricing_and_taxation.md]: We probably shouldn't be giving out tax advice:
> For instance, in Australia customers pay a Goods and Services Tax (GST). This is basically equivalent to VAT in Europe.

File: guides/pricing_and_taxation.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+***
+
+To model these behaviours in Solidus, use the `included_in_price` boolean on the `Spree::TaxRate` model. If `included_in_price` is set to `true`, Solidus will assume VAT-style taxation for this tax rate, if it is set to `false`, it will assume Sales tax style taxation.
+
+In some cases you may need to charge one type of tax for orders falling within one zone and another type of tax for orders falling within a different zone. There are even some rare situations where you may need to charge both types of tax in the same zone. Solidus  supports all of these scenarios.
```
Comment (bbuchalter) [inline guides/pricing_and_taxation.md]: "Solidus  supports" extra space

File: guides/pricing_and_taxation.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+***
+Taxation within the United States can get exceptionally complex, with different states, counties and even cities having different taxation rates. If you are shipping interstate within the United States, we would strongly advise you to use the [Solidus Avatax](https://github.com/solidusio/solidus_avatax) extension so that you get correct tax rates.
```
Comment (bbuchalter) [inline guides/pricing_and_taxation.md]: I'd also like to see here a link to the open issue @jordan-brough is working on in https://github.com/solidusio/solidus/issues/1252 to suggest that there is more than one solution to this, and it is under active development.

File: guides/pricing_and_taxation.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+Solidus can handle several prices for the same country, and will always select the most recently updated price.
+
+## Prices in the backend
+
+In the Solidus admin backend, prices will be shown and entered with or without VAT, depending on the configuration value `Spree::Config.admin_vat_country_iso`. This has to be set to the two-letter ISO code of the country your admin thinks in. For a store that mostly sells to Germany, this would be set to `DE`. The effect of this is that now all prices in the backend will assume to have the German VAT rate included.
```
Comment (bbuchalter) [inline guides/pricing_and_taxation.md]: I believe there are open issues which talk about adding more robust pricing management in the admin. Perhaps more links to those issues to suggest further active development?

File: guides/pricing_and_taxation.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+## Examples
+
+### Sales Tax
+
+Sales tax is the default tax type for any tax rate in Solidus.
+
+Let's say you need to charge 5% additional ("Sales") tax for all items that ship to New York and 6% on clothing items that ship to Pennsylvania. This will mean you need to construct two different zones: one zone containing just the state of New York and another zone consisting of the single state of Pennsylvania.
+
+Here's another hypothetical scenario. You would like to charge 10% tax on all electronic items and 5% tax on everything else. This tax should apply to all countries in the European Union (EU). In this case you would construct just a single zone consisting of all the countries in the EU. To differentiate between "electronic" items and "other" items, use tax categories. Now setup two tax rates with the "EU" zone, and their respective tax categories. The fact that you want to charge two different rates depending on the type of good does not mean you need two zones.## Examples
```
Comment (bbuchalter) [inline guides/pricing_and_taxation.md]: `##Examples` at the end of the line.

File: guides/pricing_and_taxation.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+See the [specifications](https://github.com/solidusio/solidus/blob/master/core/spec/models/spree/variant/price_selector_spec.rb) for the standard [Price Selector](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/variant/price_selector.rb) as well as the [specifications](https://github.com/solidusio/solidus/blob/master/core/spec/models/spree/variant/pricing_options_spec.rb) for the standard [Pricing Options](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/variant/pricing_options.rb) for inspiration.
+***
+
+## Examples
+
+### Sales Tax
+
+Sales tax is the default tax type for any tax rate in Solidus.
+
+Let's say you need to charge 5% additional ("Sales") tax for all items that ship to New York and 6% on clothing items that ship to Pennsylvania. This will mean you need to construct two different zones: one zone containing just the state of New York and another zone consisting of the single state of Pennsylvania.
```
Comment (bbuchalter) [inline guides/pricing_and_taxation.md]: While I appreciate the effort here, you've previously mentioned everyone in the US should _not_ use these mechanisms to collect taxes, and instead use Solidus Avatax. Adding these examples seems to contradict that.

File: guides/pricing_and_taxation.md (hunk 7 above)
Comment (bbuchalter) [inline guides/pricing_and_taxation.md]: I thought EU was VAT only? This example seems to contradict that?

File: guides/getting_started.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
+## Getting Started with Solidus
```
Comment (bbuchalter) [inline guides/getting_started.md]: This does not belong in this PR.

Comment (mamhoff) [inline guides/getting_started.md]: True that, will rebase.

File: guides/pricing_and_taxation.md (hunk 8 above)
Comment (mamhoff) [inline guides/pricing_and_taxation.md]: It's a technical document, and describes what's possible. Something like US sales tax is possible, and that's what the documentation tells you. Zipcode-level taxation is not possible, but there might be stores that do not require it... Maybe.

**Comment by mamhoff** [comment]
I've re-ordered the guide so that the VAT/Sales tax differentiation is explained in the beginning, then Solidus' taxation system is explained, then the pricing system. 

I've also removed some confusing/repeating paragraphs.

File: guides/pricing_and_taxation.md [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+
+## Prices in the backend
+
+In the Solidus admin backend, prices will be shown and entered with or without VAT, depending on the configuration value `Spree::Config.admin_vat_country_iso`. This has to be set to the two-letter ISO code of the country your admin thinks in. For a store that mostly sells to Germany, this would be set to `DE`. The effect of this is that now all prices in the backend will assume to have the German VAT rate included.
+
+## Prices in the frontend
+
+When a customer first browses your store, we do not know which jurisdiction she lives in. We will, therefore, have to make an assumption about her whereabouts. This assumption is the `cart_tax_country_iso` property on the `Spree::Store` model - meaning that you can assume your customers to be from different countries depending on which store she browses.
+
+This way, you can have, for example, two different stores for France and Germany (`my-shop.com/fr` and `my-shop.com/de`), where each shows different prices to accustom for the different VAT rates in either country.
```
Comment (jhawthorn) [inline guides/pricing_and_taxation.md]: I think this should be

> to account for the different VAT rates in either country

Comment (tvdeyen) [inline guides/pricing_and_taxation.md]: Yes

> where each shows different prices to accustom for the different VAT rates in either country

where each store shows different prices to account different VAT rates in either country

Comment (tvdeyen) [inline guides/pricing_and_taxation.md]: Maybe we should mention the introduction about "Prices have to be displayed including VAT to the customer" requirement from above here, so the reader has a better understanding why this whole `cart_tax_country_iso`  is actually needed?

Maybe:

"Remember the requirement to display the prices including vat from above? This is how you comply to this..."

File: guides/pricing_and_taxation.md [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+***
+You can customize Solidus' pricing behaviour by creating a custom `Spree::Config.variant_price_selector_class` along with a fitting `Spree::Config.pricing_options_class`.
+See the [specifications](https://github.com/solidusio/solidus/blob/master/core/spec/models/spree/variant/price_selector_spec.rb) for the standard [Price Selector](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/variant/price_selector.rb) as well as the [specifications](https://github.com/solidusio/solidus/blob/master/core/spec/models/spree/variant/pricing_options_spec.rb) for the standard [Pricing Options](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/variant/pricing_options.rb) for inspiration.
+***
+
+## Examples
+
+### Sales Tax
+
+Let's say you need to charge 5% additional ("Sales") tax for all items that ship to New York and 6% on clothing items that ship to Pennsylvania. This will mean you need to construct two different zones: one zone containing just the state of New York and another zone consisting of the single state of Pennsylvania.
```
Comment (tvdeyen) [inline guides/pricing_and_taxation.md]: Maybe additionally explain the need of both rates have to be linked to the "Clothing" tax category?

File: guides/pricing_and_taxation.md [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+
+## Examples
```
Comment (tvdeyen) [inline guides/pricing_and_taxation.md]: These "Examples" are for Examples -> Sales Tax right? Then the heading needs to be `#### Examples`

Comment (tvdeyen) [inline guides/pricing_and_taxation.md]: I think we can just remove the heading, because you are continuing the examples section from above, no?

File: guides/pricing_and_taxation.md [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+
+Let's take an example of a sales tax situation for the United States. Imagine that we have a zone that covers all of North America and that the zone is used for a tax rate which applies a 5% tax on products with the tax category of "Clothing".
+
+If the customer purchases a single clothing item for $17.99 and they live in the United States (which is within the North America zone we defined) they are required to pay sales tax.
+
+The sales tax calculation is $17.99 x 5% for a total tax of $0.8995, which is rounded up to two decimal places, to $0.90. This tax amount is then applied to the line item as an adjustment.
```
Comment (tvdeyen) [inline guides/pricing_and_taxation.md]: > This tax amount is then applied to the line item as an adjustment.

This tax amount is then applied to the line item as an adjustment and added on top of the order total.

File: guides/pricing_and_taxation.md [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+
+If the quantity of the item is changed to 2, then the tax amount doubles: ($17.99 x 2) x 0.05 is $1.799, which is again rounded up to two decimal places, applying a tax adjustment of $1.80.
+
+Let's now assume that we have another product that's a coffee mug, which doesn't have the "Clothing" tax category applied to it. Let's also assume this product costs $13.99, and there's no default tax category set up for the system. Under these circumstances, the coffee mug will not be taxed when it's added to the order.
+
+### VAT-Style taxation
+
+Many jurisdictions have what is commonly referred to as a Value Added Tax (VAT). In these cases the tax is typically already included in the price. This means that no additional tax needs to be applied during checkout.
+
+When tax is included in the price there the adjustments to the line items do not affect the order total (unlike the sales tax case). Stores are, however, usually required to show the amount of tax the user paid.
```
Comment (tvdeyen) [inline guides/pricing_and_taxation.md]: > When tax is included in the price there the adjustments to the line items do not affect the order total

When tax is included in the price then the adjustments do not affect the order total

Comment (tvdeyen) [inline guides/pricing_and_taxation.md]: > Stores are, however, usually required to show the amount of tax the user paid.

Stores are, however, usually required to show the amount of tax the user paid. That's why Solidus lists all adjustments below the item total on the checkout summary page.

File: guides/pricing_and_taxation.md [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+
+Now let's assume an additional tax rate of 10% on a "Consumer Electronics" tax category. When we add a product with this tax category to our order with a price of &pound;16.99, there will be a second adjustment added to the order, with a calculated total of &pound;16.99 x 10%, which is &pound;1.699. Rounded up, it's &pound;1.70.
+
+### Different tax rates by tax category for the same zone
+
+Here's another hypothetical scenario. You would like to charge 10% tax on all electronic items and 5% tax on everything else. This tax should apply to all countries in the European Union (EU). In this case you would construct just a single zone consisting of all the countries in the EU. To differentiate between "electronic" items and "other" items, use tax categories. Now setup two tax rates with the "EU" zone, and their respective tax categories. The fact that you want to charge two different rates depending on the type of good does not mean you need two zones.
```
Comment (tvdeyen) [inline guides/pricing_and_taxation.md]: > Here's another hypothetical scenario

Aren't you explaining a MOSS scenario here? So, this is not hypothetical at all, no?

File: guides/pricing_and_taxation.md [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+
+Per default Solidus assumes that all products are without tax. Only if a product has a tax category set, or there is a global default tax category set, will products be taxed. The tax category is specific to taxation, and is normally never seen by the user so you could call it something generic like "Taxable Goods." If you wish to tax certain products at different rates, however, then you will want to choose something more descriptive (ex. "Clothing.").
+
+### Shipping vs. Billing Address
+
+Most tax jurisdictions base the tax on the shipping address of where the order is being shipped to. So in these cases the shipping address is used when determining the tax zone. Solidus does, however, allow you to use the billing address instead.
+
+To determine tax based on billing address instead of shipping address you will need to set the `Spree::Config[:tax_using_ship_address]` preference to `false`.
```
Comment (cbrunsdon) [inline guides/pricing_and_taxation.md]: Is there a good use case for this? Or is it just for like... digital goods or something? I would say bringing attention to it might not be worth it here unless its something we want to be keeping long term.

Comment (mamhoff) [inline guides/pricing_and_taxation.md]: No, I haven't heard of a single instance where anyone set this to `false`.  I'll remove it from the guide and send a PR deprecating it.

Comment (mamhoff) [inline guides/pricing_and_taxation.md]: I've asked a number of times (at SolidusConf, in the Slack channel...). I've tried finding out where it comes from, and found this commit: https://github.com/spree/spree/commit/d3ee56193d6fc4e40b9d84f6cbd154b958d96477 - there's no explanation as to why it got introduced.

Comment (mamhoff) [inline guides/pricing_and_taxation.md]: So apparently there is edge cases in the US where for some orders, depending on where the stock location is, and depending on where the customer is located. See [this blog post](http://blog.taxjar.com/use-billing-address-shipping-address-calculating-sales-tax/) (I did not understand when it actually happens). 

However, this configuration option is *global*. I'm still for killing it; in case you need that behaviour globally, it's easy to override `Spree::Order#tax_address`.

Comment (mamhoff) [inline guides/pricing_and_taxation.md]: Added it back in.
